src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -340,5 +412,349 @@ private static ICollection<object> PopulateHashTableFromJArray(JArray list, out
+        private static object AddPsProperties(object psObj, object obj, int depth, bool isPurePSObj, bool isCustomObj, in ConvertToJsonContext context)
+        {
+            PSObject pso = psObj as PSObject;
+
+            if (pso == null)
+                return obj;
+
+            // when isPurePSObj is true, the obj is guaranteed to be a string converted by LanguagePrimitives
+            if (isPurePSObj)
+                return obj;
+
+            bool wasDictionary = true;
+            IDictionary dict = obj as IDictionary;
+
+            if (dict == null)
+            {
+                wasDictionary = false;
+                dict = new Dictionary<string, object>();
+                dict.Add("value", obj);
+            }
+
+            AppendPsProperties(pso, dict, depth, isCustomObj, in context);
+
+            if (wasDictionary == false && dict.Count == 1)
+                return obj;
+
+            return dict;
+        }
+
+        /// <summary>
+        /// Append to a dictionary any properties that might have been added to an object (via PSObject) through the Add-Member cmdlet.
+        /// If the passed in object is a custom object (not a simple object, not a dictionary, not a list, get processed in ProcessCustomObject method),
+        /// we also take Adapted properties into account. Otherwise, we only consider the Extended properties.
+        /// When the object is a pure PSObject, it also gets processed in "ProcessCustomObject" before reaching this method, so we will
+        /// iterate both extended and adapted properties for it. Since it's a pure PSObject, there will be no adapted properties.
+        /// </summary>
+        /// <param name="psObj">The containing PSObject, or null if the base object was not contained in a PSObject.</param>
+        /// <param name="receiver">The dictionary to which any additional properties will be appended.</param>
+        /// <param name="depth">The current depth into the object graph.</param>
+        /// <param name="isCustomObject">The processed object is a custom object.</param>
+        /// <param name="context">The context for the operation.</param>
+        private static void AppendPsProperties(PSObject psObj, IDictionary receiver, int depth, bool isCustomObject, in ConvertToJsonContext context)
+        {
+            // serialize only Extended and Adapted properties..
+            PSMemberInfoCollection<PSPropertyInfo> srcPropertiesToSearch =
+                new PSMemberInfoIntegratingCollection<PSPropertyInfo>(psObj,
+                    isCustomObject ? PSObject.GetPropertyCollection(PSMemberViewTypes.Extended | PSMemberViewTypes.Adapted) :
+                    PSObject.GetPropertyCollection(PSMemberViewTypes.Extended));
+
+            foreach (PSPropertyInfo prop in srcPropertiesToSearch)
+            {
+                object value = null;
+                try
+                {
+                    value = prop.Value;
+                }
+                catch (Exception)
```

Comment:
It is not best practice to catch all exceptions.

Reply:
I don't want to break the current behavior to throw out any exceptions in this case.
It's also the behavior for formatting code -- when accessing a property throws exception, ignore the exception and use null for the value.